numcodecs/tests/test_compat.py

```diff
@@ -2,13 +2,22 @@
 from __future__ import absolute_import, print_function, division
 import array
 
+import pytest
 
 import numpy as np
 
 
 from numcodecs.compat import buffer_tobytes
 
 
+def test_buffer_tobytes_raises():
+    a = np.array([u'Xin chào thế giới'], dtype=object)
+    with pytest.raises(ValueError):
+        buffer_tobytes(a)
+    with pytest.raises(ValueError):
+        buffer_tobytes(memoryview(a))
```

Comment:
Nice to test this, didn't realise you could get a memoryview of an object array.

Reply:
Yeah, not sure how useful `object` type `memoryview`s are, but they do seem to be allowed.